ui/components/component-library/text/text.js

```diff
@@ -0,0 +1,187 @@
+import React from 'react';
+import classnames from 'classnames';
+import PropTypes from 'prop-types';
+import {
+  COLORS,
+  FONT_WEIGHT,
+  FONT_STYLE,
+  TEXT_ALIGN,
+  TEXT,
+  OVERFLOW_WRAP,
+} from '../../../helpers/constants/design-system';
+import Box, { MultipleSizesAndAuto } from '../../ui/box';
+
+export const ValidColors = [
+  COLORS.TEXT_DEFAULT,
+  COLORS.TEXT_ALTERNATIVE,
+  COLORS.TEXT_MUTED,
+  COLORS.OVERLAY_INVERSE,
+  COLORS.PRIMARY_DEFAULT,
+  COLORS.PRIMARY_INVERSE,
+  COLORS.SECONDARY_DEFAULT,
+  COLORS.SECONDARY_INVERSE,
+  COLORS.ERROR_DEFAULT,
+  COLORS.ERROR_INVERSE,
+  COLORS.SUCCESS_DEFAULT,
+  COLORS.SUCCESS_INVERSE,
+  COLORS.WARNING_INVERSE,
+  COLORS.INFO_DEFAULT,
+  COLORS.INFO_INVERSE,
+];
+
+export const ValidTags = [
+  'dd',
+  'div',
+  'dt',
+  'em',
+  'h1',
+  'h2',
+  'h3',
+  'h4',
+  'h5',
+  'h6',
+  'li',
+  'p',
+  'span',
+  'strong',
+  'ul',
+  'label',
+];
+
+export const Text = ({
+  variant = TEXT.BODY_MD,
+  color = COLORS.TEXT_DEFAULT,
+  fontWeight,
+  fontStyle,
+  align,
+  overflowWrap,
+  ellipsis,
+  title,
+  as,
+  margin,
+  marginTop,
+  marginRight,
+  marginBottom,
+  marginLeft,
+  className,
+  children,
+  ...props
+}) => {
+  let Tag = as ?? variant;
+  let strongTagFontWeight;
+
+  if (Tag === 'strong') {
+    strongTagFontWeight = FONT_WEIGHT.BOLD;
+  }
+
+  const computedClassName = classnames(
+    'text',
+    className,
+    `text--${variant}`,
+
+    (strongTagFontWeight || fontWeight) &&
+      `text--weight-${strongTagFontWeight || fontWeight}`,
+    fontStyle && `text--style-${fontStyle}`,
+    { [`text--ellipsis`]: Boolean(ellipsis) },
+    {
+      [`text--align-${align}`]: Boolean(align),
+      [`text--color-${color}`]: Boolean(color),
+      [`text--overflowwrap-${overflowWrap}`]: Boolean(overflowWrap),
+    },
+  );
+
+  // // Set a default tag based on variant
+  const splitTag = Tag.split('-')[0];
+  if (splitTag === 'body') {
+    Tag = 'p';
+  } else if (splitTag === 'heading') {
+    Tag = 'h2';
+  } else if (splitTag === 'display') {
+    Tag = 'h1';
+  }
+
+  return (
+    <Box
+      {...{
+        margin,
+        marginTop,
+        marginRight,
+        marginBottom,
+        marginLeft,
+        ...props,
+      }}
+    >
+      {(boxClassName) => (
+        <Tag
+          className={classnames(boxClassName, computedClassName)}
+          title={title}
+        >
+          {children}
+        </Tag>
+      )}
+    </Box>
+  );
+};
+
+Text.propTypes = {
+  /**
+   * The variation of font types of the Text component (display, heading, body)
```

Comment:
nit: Description suggestion
```suggestion
   * The variation of font styles including sizes and weights of the Text component (display, heading, body)
```